test/Microsoft.VisualStudio.Sdk.TestFramework.Tests/Mocks/GreetMock.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using static System.Reflection.BindingFlags;
+
+public class GreetMock : IGreet, IMockServiceWithClientCallback
+{
+    public event EventHandler<string>? GreetingSent;
+
+    public event EventHandler? OperationComplete;
+
+    public object ClientCallback { get; set; } = new object();
+
+    public async ValueTask<string> MakeGreeting(CancellationToken cancellationToken)
+    {
+#pragma warning disable CS8605
+        var name = await (ValueTask<string>)typeof(ISayName).GetMethod("SayName")?.Invoke(this.ClientCallback, null);
+#pragma warning restore CS8605
```

Comment:
Reflection isn't necessary here. Just a type cast.

```suggestion
        string name = await ((ISayName)this.ClientCallback).SayName();
```

Reply:
Great, thank you. I was struggling a bit with how to invoke the call.